exchanges/deribit/deribit.go

```diff
@@ -0,0 +1,2797 @@
+	createSubAccount                  = "private/create_subaccount"
+	disableAPIKey                     = "private/disable_api_key"
+	editAPIKey                        = "private/edit_api_key"
+	enableAffiliateProgram            = "private/enable_affiliate_program"
+	enableAPIKey                      = "private/enable_api_key"
+	getAccessLog                      = "private/get_access_log"
+	getAccountSummary                 = "private/get_account_summary"
+	getAffiliateProgramInfo           = "private/get_affiliate_program_info"
+	getEmailLanguage                  = "private/get_email_language"
+	getNewAnnouncements               = "private/get_new_announcements"
+	getPrivatePortfolioMargins        = "private/get_portfolio_margins"
+	getPosition                       = "private/get_position"
+	getPositions                      = "private/get_positions"
+	getSubAccounts                    = "private/get_subaccounts"
+	getSubAccountDetails              = "private/get_subaccounts_details"
+	getTransactionLog                 = "private/get_transaction_log"
+	getUserLocks                      = "private/get_user_locks"
+	listAPIKeys                       = "private/list_api_keys"
+	listCustodyAccounts               = "private/list_custody_accounts"
+	removeAPIKey                      = "private/remove_api_key"
+	removeSubAccount                  = "private/remove_subaccount"
+	resetAPIKey                       = "private/reset_api_key"
+	setAnnouncementAsRead             = "private/set_announcement_as_read"
+	setEmailForSubAccount             = "private/set_email_for_subaccount"
+	setEmailLanguage                  = "private/set_email_language"
+	setSelfTradingConfig              = "private/set_self_trading_config"
+	toggleNotificationsFromSubAccount = "private/toggle_notifications_from_subaccount"
+	togglePortfolioMargining          = "private/toggle_portfolio_margining"
+	toggleSubAccountLogin             = "private/toggle_subaccount_login"
+
+	// Combo Books Endpoints
+	getComboDetails = "public/get_combo_details"
+	getComboIDs     = "public/get_combo_ids"
+	getCombos       = "public/get_combos"
+	createCombos    = "private/create_combo"
+
+	// Block Trades Endpoints
+	executeBlockTrades             = "private/execute_block_trade"
+	getBlockTrades                 = "private/get_block_trade"
+	getLastBlockTradesByCurrency   = "private/get_last_block_trades_by_currency"
+	invalidateBlockTradesSignature = "private/invalidate_block_trade_signature"
+	movePositions                  = "private/move_positions"
+	simulateBlockPosition          = "private/simulate_block_trade"
+	verifyBlockTrades              = "private/verify_block_trade"
+
+	// roles
+
+	roleMaker = "maker"
+	roleTaker = "taker"
+)
+
+// GetBookSummaryByCurrency gets book summary data for currency requested
+func (d *Deribit) GetBookSummaryByCurrency(ctx context.Context, ccy, kind string) ([]BookSummaryData, error) {
+	if ccy == "" {
```

Comment:
Where we can test these conditions, we should always add test coverage for it. As an example:

```
_, err = d.GetBookSummaryByCurrency(context.Background(), "", "future")
assert.ErrorIs(t, err, errInvalidCurrency)
```